system-contracts/contracts/precompiles/EcAdd.yul

```diff
@@ -291,150 +65,27 @@ object "EcAdd" {
-                // P + Infinity = P
-
-                // Ensure that the coordinates are between 0 and the field order.
-                if or(iszero(isOnFieldOrder(x1)), iszero(isOnFieldOrder(y1))) {
-                    burnGas()
-                }
-
-                let m_x1 := intoMontgomeryForm(x1)
-                let m_y1 := intoMontgomeryForm(y1)
-
-                // Ensure that the point is in the curve (Y^2 = X^3 + 3).
-                if iszero(pointIsInCurve(m_x1, m_y1)) {
-                    burnGas()
-                }
-
-                // We just need to go into the Montgomery form to perform the
-                // computations in pointIsInCurve, but we do not need to come back.
-
-                mstore(0, x1)
-                mstore(32, y1)
-                return(0, 64)
-            }
-
-            // Ensure that the coordinates are between 0 and the field order.
-            if or(iszero(isOnFieldOrder(x1)), iszero(isOnFieldOrder(y1))) {
-                burnGas()
-            }
-
-            // Ensure that the coordinates are between 0 and the field order.
-            if or(iszero(isOnFieldOrder(x2)), iszero(isOnFieldOrder(y2))) {
-                burnGas()
-            }
-
-            // There's no need for transforming into Montgomery form
-            // for this case.
-            if and(eq(x1, x2), eq(submod(0, y1, P()), y2)) {
-                // P + (-P) = Infinity
-
-                let m_x1 := intoMontgomeryForm(x1)
-                let m_y1 := intoMontgomeryForm(y1)
-                let m_x2 := intoMontgomeryForm(x2)
-                let m_y2 := intoMontgomeryForm(y2)
-
-                // Ensure that the points are in the curve (Y^2 = X^3 + 3).
-                if or(iszero(pointIsInCurve(m_x1, m_y1)), iszero(pointIsInCurve(m_x2, m_y2))) {
-                    burnGas()
-                }
+            // We conduct all validations inside the precompileCall
 
-                // We just need to go into the Montgomery form to perform the
-                // computations in pointIsInCurve, but we do not need to come back.
+            mstore(0, x1)
+            mstore(32, y1)
+            mstore(64, x2)
+            mstore(96, y2)
 
-                mstore(0, 0)
-                mstore(32, 0)
-                return(0, 64)
-            }
+            let precompileParams := unsafePackPrecompileParams(
+                0, // input offset in words
+                4, // input length in words (x1, y1, x2, y2)
+                0, // output offset in words
+                2, // output length in words (x, y)
+                0  // No special meaning, ecadd circuit doesn't check this value
+            )
+            let gasToPay := ECADD_GAS_COST()
 
-            if and(eq(x1, x2), and(iszero(eq(y1, y2)), iszero(eq(y1, submod(0, y2, P()))))) {
-                burnGas()
+            let success := precompileCall(precompileParams, gasToPay)
+            if not(success) {
```

Comment:
While this is technically works, `iszero` is used usually instead of `not` in yul to convert number to boolean

Reply:
I changed it to improve readability, I can revert it back to use `iszero` though - what do you think?

Reply:
I believe that it is better to keep it `iszero` as `not` is confusing and may introduce issues when devs will rewrite the contract. (`not` in Yul may be confused with `!` [not operation] in Solidity, but it is different)

Reply:
oh, okay cool, then I'll change it back